tests/Unit/DataStructures/DataBox/Test_DataBoxDocumentation.cpp

```diff
@@ -0,0 +1,512 @@
+  }
+};
+/// [intended_mutation2]
+
+/// [my_first_action]
+template <typename Mutator>
+struct MyFirstAction{
+  template<typename DbTagsList>
+  static void apply(
+    const gsl::not_null<db::DataBox<DbTagsList>*> refined_databox) noexcept {
+    db::mutate_apply<typename Mutator::mutate_tags,
+                     typename Mutator::argument_tags>(
+    Mutator{}, refined_databox);
+  }
+};
+/// [my_first_action]
+
+} // namespace
+
+SPECTRE_TEST_CASE("Unit.DataStructures.DataBox.Documentation",
+                  "[Unit][DataStructures]") {
+const double velocity = 4.0;
+const double radius = 2.0;
+const double density = 0.5;
+const double volume = 10.0;
+const double mass = mass_compute(density, volume);
+const double acceleration = acceleration_compute(velocity, radius);
+const double force = force_compute(velocity, radius, density, volume);
+CHECK(mass == density*volume);
+CHECK(acceleration == velocity * velocity / radius);
+CHECK(force == mass * acceleration);
+failed_acceleration();
+CHECK(force == working_without_databoxes_1());
+CHECK(force == working_without_databoxes_2());
+CHECK(force == std_map_databox_1());
+CHECK(force == std_map_databox_2());
+std::map<std::string, double> naive_databox;
+naive_databox["Velocity"] = velocity;
+naive_databox["Radius"] = radius;
+naive_databox["Density"] = density;
+naive_databox["Volume"] = volume;
+CHECK(mass == mass_compute(naive_databox));
+CHECK(acceleration == acceleration_compute(naive_databox));
+CHECK(force == force_compute(naive_databox));
+CHECK(std_tuple_databox_example());
+std::tuple<std::pair<Velocity,double>,
+           std::pair<Radius, double>, std::pair<Density, double>,
+           std::pair<Volume, double>> sophomore_databox =
+  std::make_tuple(std::make_pair(Velocity{}, 4.0),
+                  std::make_pair(Radius{}, 2.0), std::make_pair(Density{}, 0.5),
+                  std::make_pair(Volume{}, 10.0));
+CHECK(mass == mass_compute(sophomore_databox));
+CHECK(acceleration == acceleration_compute(sophomore_databox));
+CHECK(force == force_compute(sophomore_databox));
+CHECK(force == std_tuple_databox_1());
+CHECK(force == std_tuple_databox_2());
+tuples::TaggedTuple<Velocity2, Radius2, Density2, Volume2> junior_databox{
+  velocity, radius, density, volume};
+CHECK(mass ==  mass_compute(junior_databox));
+CHECK(acceleration == acceleration_compute(junior_databox));
+CHECK(force == force_compute(junior_databox));
+CHECK(force == tagged_tuple_databox_1());
+CHECK(force == refined_databox_1());
+const auto refined_databox = db::create<
+    db::AddSimpleTags<Velocity3, Radius3, Density3, Volume3>,
+    db::AddComputeTags<ForceCompute>>(
+      velocity, radius, density, volume);
+CHECK(force == db::get<Force>(refined_databox));
+
+/// [time_dep_databox]
+auto time_dependent_databox = db::create<
+    db::AddSimpleTags<
+      Time, TimeStep, EarthGravity, FallingSpeed>>(0.0, 0.1, -9.8, -10.0);
+db::mutate_apply<
+  //MutateTags
```

Comment:
Could you add `db::MutateTags` and `db::ArgumentTags` to `src/DataStructures/DataBox/DataBox.hpp` at about line 1227? The implementation should be the same as the ones for `db::AddSimpleTags`, etc.